Data/Yaml.hs

```diff
@@ -190,6 +195,14 @@ decodeEither' = either Left (either (Left . AesonException) Right)
               . decodeHelper
               . Y.decode
 
+-- | A version of 'decodeEither'' lifted to MonadThrow
+decodeThrow :: (MonadThrow m, FromJSON a) => ByteString -> m a
+decodeThrow = either throwM return . decodeEither'
+
+-- | A version of 'decodeFileEither' lifted to MonadIO, MonadThrow
+decodeFileThrow :: (MonadIO m, MonadThrow m, FromJSON a) => FilePath -> m a
```

Comment:
Is there a reason to put this in `MonadThrow`? I've been leaning towards the combo `MonadIO`+`MonadThrow` to be an anti-pattern, as it's ambiguous about where the exception could be thrown.

Reply:
It won't compile without the MonadThrow constraint

```
/home/lc/Source/github.com/snoyberg/yaml/Data/Yaml.hs:208:56: error:
    • Could not deduce (MonadThrow m) arising from a use of ‘throwM’
      from the context: (MonadIO m, FromJSON a)
        bound by the type signature for:
                   decodeFileThrow :: (MonadIO m, FromJSON a) => FilePath -> m a
        at Data/Yaml.hs:207:1-61
      Possible fix:
        add (MonadThrow m) to the context of
          the type signature for:
            decodeFileThrow :: (MonadIO m, FromJSON a) => FilePath -> m a
    • In the first argument of ‘either’, namely ‘throwM’
      In the second argument of ‘(>=>)’, namely ‘either throwM return’
      In the expression:
        liftIO . decodeFileEither >=> either throwM return
```

Reply:
You can change the definition slightly:

```haskell
decodeFileThrow fp = liftIO $ decodeFileEither fp >>= either throwIO return
```